Dockerfile

```diff
@@ -52,4 +41,26 @@ WORKDIR /usr/src/app/OpenOversight
 
 COPY OpenOversight .
 
+# Development Target
+FROM base as development
+ARG REQUIREMENTS_FILE
+
+RUN apt-get install -y firefox-esr xvfb
+RUN pip3 install -r /usr/src/app/${REQUIREMENTS_FILE}
+
+ENV GECKODRIVER_VERSION="v0.26.0"
+ENV GECKODRIVER_SHA=d59ca434d8e41ec1e30dd7707b0c95171dd6d16056fb6db9c978449ad8b93cc0
+ENV GECKODRIVER_BASE_URL="https://github.com/mozilla/geckodriver/releases/download"
+RUN wget ${GECKODRIVER_BASE_URL}/${GECKODRIVER_VERSION}/geckodriver-${GECKODRIVER_VERSION}-linux64.tar.gz
+RUN echo "${GECKODRIVER_SHA}  geckodriver-${GECKODRIVER_VERSION}-linux64.tar.gz" | sha256sum --check -
+RUN tar -xzf geckodriver-v0.26.0-linux64.tar.gz -C /usr/bin
+
+RUN apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
```

Comment:
Unfortunately, due to the way that [Docker layers work](https://docs.docker.com/storage/storagedriver/), this step only adds to the size of the image. These clean steps have to happen in the same step as the `apt install` steps, otherwise the deltas of each `RUN` step get written into the image.

Reply:
Sorry I don't think I'm explaining this well haha. The best way to think about this is that each `RUN` step adds onto the image size. So the state at the end of a `RUN` step is what gets written to disk, therefore we have to do the cleanup in the same step as the install.

Reply:
I think that makes sense! As you can see, I know next to nothing about docker lolol

The only way I can see avoiding this would be to push the apt-get `RUN` down into the development/production targets. Does that make sense?

Reply:
Yea it does! Alternatively, we could do two `RUN` steps - one in the builder image that has the packages both containers use, and another in the dev target that adds the XVFB requirements. Both `RUN` steps would need to do the `clean` and `rm -rf /var/lib/apt/lists/*` steps

Reply:
That works!